fix current compiler error

running `cargo run` in the `api` directory errors on:

```
error[E0063]: missing fields `audio_bg` and `audio_effects` in initializer of `shared::domain::jig::Jig`
   --> src\db\jig.rs:135:20
    |
135 |         .map(|row| Jig {
    |                    ^^^ missing `audio_bg` and `audio_effects`

error[E0063]: missing fields `audio_bg` and `audio_effects` in initializer of `shared::domain::jig::Jig`
   --> src\db\jig.rs:201:16
    |
201 |     .map(|row| Jig {
    |                ^^^ missing `audio_bg` and `audio_effects`

error[E0063]: missing fields `audio_bg` and `audio_effects` in initializer of `shared::domain::jig::Jig`
   --> src\db\jig.rs:387:19
    |
387 |     .map_ok(|row| Jig {
    |                   ^^^ missing `audio_bg` and `audio_effects`
```

--- a/backend/api/fixtures/8_jig.sql
+++ b/backend/api/fixtures/8_jig.sql
@@ -1,6 +1,6 @@
-insert into jig (id, creator_id, author_id, created_at, language, description, is_public)
+insert into jig (id, creator_id, author_id, created_at, language, description, is_public, audio_background, audio_feedback_positive, audio_feedback_negative)
 values
-    ('0cc084bc-7c83-11eb-9f77-e3218dffb008', '1f241e1b-b537-493f-a230-075cb16315be', '1f241e1b-b537-493f-a230-075cb16315be', '2021-03-04 00:46:26.134651+00', 'en', 'test description', true);
+    ('0cc084bc-7c83-11eb-9f77-e3218dffb008', '1f241e1b-b537-493f-a230-075cb16315be', '1f241e1b-b537-493f-a230-075cb16315be', '2021-03-04 00:46:26.134651+00', 'en', 'test description', true, null, array[0, 1], array[0, 1, 4]);
 
 
 insert into jig_module (jig_id, id, index, kind, contents, created_at, is_complete)

--- a/backend/api/migrations/20210617221637_jig-settings-audio-select.sql
+++ b/backend/api/migrations/20210617221637_jig-settings-audio-select.sql
@@ -1,0 +1,4 @@
+alter table jig
+    add column audio_background        smallint, -- null allowed
+    add column audio_feedback_negative smallint[] not null default '{}',
+    add column audio_feedback_positive smallint[] not null default '{}';

--- a/backend/api/sqlx-data.json
+++ b/backend/api/sqlx-data.json
@@ -1,5 +1,147 @@
 {
   "db": "PostgreSQL",
+  "00a3c661403e42c0871ad7b8978ef3980e59853b9106a253efb956a89a24c9f6": {
+    "query": "\nselect  \n    id as \"id: JigId\",\n    display_name,\n    creator_id,\n    author_id,\n    publish_at,\n    updated_at,\n    language,\n    description,\n    is_public,\n    direction as \"direction: TextDirection\",\n    display_score,\n    theme as \"theme: ThemeId\",\n    audio_background as \"audio_background!: Option<AudioBackground>\",\n    array(select row(unnest(audio_feedback_positive))) as \"audio_feedback_positive!: Vec<(AudioFeedbackPositive,)>\", -- TODO: fix ugly!\n    array(select row(unnest(audio_feedback_negative))) as \"audio_feedback_negative!: Vec<(AudioFeedbackNegative,)>\",\n    array(\n        select row (id, kind)\n        from jig_module\n        where jig_id = jig.id\n        order by \"index\"\n    ) as \"modules!: Vec<(ModuleId, ModuleKind)>\",\n    array(select row(goal_id) from jig_goal where jig_id = jig.id) as \"goals!: Vec<(GoalId,)>\",\n    array(select row(category_id) from jig_category where jig_id = jig.id) as \"categories!: Vec<(CategoryId,)>\",\n    array(select row(affiliation_id) from jig_affiliation where jig_id = jig.id) as \"affiliations!: Vec<(AffiliationId,)>\",\n    array(select row(age_range_id) from jig_age_range where jig_id = jig.id) as \"age_ranges!: Vec<(AgeRangeId,)>\",\n    array(select row(id) from jig_additional_resource where jig_id = jig.id) as \"additional_resources!: Vec<(AdditionalResourceId,)>\"\nfrom jig\nwhere \n    publish_at < now() is not distinct from $1 or $1 is null\n    and author_id is not distinct from $3 or $3 is null\norder by coalesce(updated_at, created_at) desc\nlimit 20 offset 20 * $2\n",
+    "describe": {
+      "columns": [
+        {
+          "ordinal": 0,
+          "name": "id: JigId",
+          "type_info": "Uuid"
+        },
+        {
+          "ordinal": 1,
+          "name": "display_name",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 2,
+          "name": "creator_id",
+          "type_info": "Uuid"
+        },
+        {
+          "ordinal": 3,
+          "name": "author_id",
+          "type_info": "Uuid"
+        },
+        {
+          "ordinal": 4,
+          "name": "publish_at",
+          "type_info": "Timestamptz"
+        },
+        {
+          "ordinal": 5,
+          "name": "updated_at",
+          "type_info": "Timestamptz"
+        },
+        {
+          "ordinal": 6,
+          "name": "language",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 7,
+          "name": "description",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 8,
+          "name": "is_public",
+          "type_info": "Bool"
+        },
+        {
+          "ordinal": 9,
+          "name": "direction: TextDirection",
+          "type_info": "Int2"
+        },
+        {
+          "ordinal": 10,
+          "name": "display_score",
+          "type_info": "Bool"
+        },
+        {
+          "ordinal": 11,
+          "name": "theme: ThemeId",
+          "type_info": "Int2"
+        },
+        {
+          "ordinal": 12,
+          "name": "audio_background!: Option<AudioBackground>",
+          "type_info": "Int2"
+        },
+        {
+          "ordinal": 13,
+          "name": "audio_feedback_positive!: Vec<(AudioFeedbackPositive,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 14,
+          "name": "audio_feedback_negative!: Vec<(AudioFeedbackNegative,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 15,
+          "name": "modules!: Vec<(ModuleId, ModuleKind)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 16,
+          "name": "goals!: Vec<(GoalId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 17,
+          "name": "categories!: Vec<(CategoryId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 18,
+          "name": "affiliations!: Vec<(AffiliationId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 19,
+          "name": "age_ranges!: Vec<(AgeRangeId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 20,
+          "name": "additional_resources!: Vec<(AdditionalResourceId,)>",
+          "type_info": "RecordArray"
+        }
+      ],
+      "parameters": {
+        "Left": [
+          "Bool",
+          "Int4",
+          "Uuid"
+        ]
+      },
+      "nullable": [
+        false,
+        false,
+        true,
+        true,
+        true,
+        true,
+        false,
+        false,
+        false,
+        false,
+        false,
+        false,
+        true,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null
+      ]
+    }
+  },
   "029540ace163666c276ac5f1a3f9eff2b277c6d73d35cbc20cab7d0b3ce73698": {
     "query": "\n        select style_id as \"id: AnimationStyleId\", display_name, animation_style.created_at, updated_at\n        from animation_style\n            left join style on animation_style.style_id = style.id\n        order by index\n        ",
     "describe": {
@@ -346,27 +488,6 @@
       ]
     }
   },
-  "0f70df2423515458cea70695c94e7732340dfd0f68bb4979c5802d13c968f279": {
-    "query": "\ninsert into jig (display_name, parents, creator_id, author_id, language, description, direction, display_score, theme)\nselect display_name, array_append(parents, id), $2 as creator_id, $2 as author_id, language, description, direction, display_score, theme\nfrom jig\nwhere id = $1\nreturning id\n        ",
-    "describe": {
-      "columns": [
-        {
-          "ordinal": 0,
-          "name": "id",
-          "type_info": "Uuid"
-        }
-      ],
-      "parameters": {
-        "Left": [
-          "Uuid",
-          "Uuid"
-        ]
-      },
-      "nullable": [
-        false
-      ]
-    }
-  },
   "1011937f14e650f4468f544ba25371e055e8bef4b55f3b6183725fe9309d28d4": {
     "query": "\ninsert into jig_affiliation(jig_id, affiliation_id)\nselect $1, affiliation_id from jig_affiliation where jig_id = $2\n",
     "describe": {
@@ -632,6 +753,27 @@
       "nullable": []
     }
   },
+  "1be4378bc2bd7b4e9b05583dc0daec0fb1daba6ffbc3c0cbdd37397f73150e1a": {
+    "query": "\ninsert into jig (display_name, parents, creator_id, author_id, language, description, direction, display_score, theme,\n                 audio_background, audio_feedback_positive, audio_feedback_negative)\nselect display_name,\n       array_append(parents, id),\n       $2 as creator_id,\n       $2 as author_id,\n       language,\n       description,\n       direction,\n       display_score,\n       theme,\n       audio_background,\n       audio_feedback_positive,\n       audio_feedback_negative\nfrom jig\nwhere id = $1\nreturning id\n        ",
+    "describe": {
+      "columns": [
+        {
+          "ordinal": 0,
+          "name": "id",
+          "type_info": "Uuid"
+        }
+      ],
+      "parameters": {
+        "Left": [
+          "Uuid",
+          "Uuid"
+        ]
+      },
+      "nullable": [
+        false
+      ]
+    }
+  },
   "1c7400d27c33a6b7204302b5cf2ced8fb445de00304be5b424133f3b7afbf1cb": {
     "query": "insert into user_auth_basic (user_id, email, password) values ($1, $2::text, $3)",
     "describe": {
@@ -836,128 +978,17 @@
       "nullable": []
     }
   },
-  "23db26987f1645238822dff897d95ba0360e68792a408c7cd65608738399c3a4": {
-    "query": "\nselect  \n    id as \"id: JigId\",\n    display_name,\n    creator_id,\n    author_id,\n    publish_at,\n    updated_at,\n    language,\n    description,\n    is_public,\n    direction as \"direction: TextDirection\",\n    display_score,\n    theme as \"theme: ThemeId\",\n    array(\n        select row (id, kind)\n        from jig_module\n        where jig_id = jig.id\n        order by \"index\"\n    ) as \"modules!: Vec<(ModuleId, ModuleKind)>\",\n    array(select row(goal_id) from jig_goal where jig_id = jig.id) as \"goals!: Vec<(GoalId,)>\",\n    array(select row(category_id) from jig_category where jig_id = jig.id) as \"categories!: Vec<(CategoryId,)>\",\n    array(select row(affiliation_id) from jig_affiliation where jig_id = jig.id) as \"affiliations!: Vec<(AffiliationId,)>\",\n    array(select row(age_range_id) from jig_age_range where jig_id = jig.id) as \"age_ranges!: Vec<(AgeRangeId,)>\",\n    array(select row(id) from jig_additional_resource where jig_id = jig.id) as \"additional_resources!: Vec<(AdditionalResourceId,)>\"\nfrom jig\nwhere \n    publish_at < now() is not distinct from $1 or $1 is null\n    and author_id is not distinct from $3 or $3 is null\norder by coalesce(updated_at, created_at) desc\nlimit 20 offset 20 * $2\n",
+  "2493076ad606463154fd13635bd6c37e9ef3044edc2ef2c0214cff99ed507200": {
+    "query": "\nupdate jig\nset audio_background = $2, updated_at = now()\nwhere id = $1 and $2 is distinct from audio_background\n            ",
     "describe": {
-      "columns": [
-        {
-          "ordinal": 0,
-          "name": "id: JigId",
-          "type_info": "Uuid"
-        },
-        {
-          "ordinal": 1,
-          "name": "display_name",
-          "type_info": "Text"
-        },
-        {
-          "ordinal": 2,
-          "name": "creator_id",
-          "type_info": "Uuid"
-        },
-        {
-          "ordinal": 3,
-          "name": "author_id",
-          "type_info": "Uuid"
-        },
-        {
-          "ordinal": 4,
-          "name": "publish_at",
-          "type_info": "Timestamptz"
-        },
-        {
-          "ordinal": 5,
-          "name": "updated_at",
-          "type_info": "Timestamptz"
-        },
-        {
-          "ordinal": 6,
-          "name": "language",
-          "type_info": "Text"
-        },
-        {
-          "ordinal": 7,
-          "name": "description",
-          "type_info": "Text"
-        },
-        {
-          "ordinal": 8,
-          "name": "is_public",
-          "type_info": "Bool"
-        },
-        {
-          "ordinal": 9,
-          "name": "direction: TextDirection",
-          "type_info": "Int2"
-        },
-        {
-          "ordinal": 10,
-          "name": "display_score",
-          "type_info": "Bool"
-        },
-        {
-          "ordinal": 11,
-          "name": "theme: ThemeId",
-          "type_info": "Int2"
-        },
-        {
-          "ordinal": 12,
-          "name": "modules!: Vec<(ModuleId, ModuleKind)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 13,
-          "name": "goals!: Vec<(GoalId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 14,
-          "name": "categories!: Vec<(CategoryId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 15,
-          "name": "affiliations!: Vec<(AffiliationId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 16,
-          "name": "age_ranges!: Vec<(AgeRangeId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 17,
-          "name": "additional_resources!: Vec<(AdditionalResourceId,)>",
-          "type_info": "RecordArray"
-        }
-      ],
+      "columns": [],
       "parameters": {
         "Left": [
-          "Bool",
-          "Int4",
-          "Uuid"
+          "Uuid",
+          "Int2"
         ]
       },
-      "nullable": [
-        false,
-        false,
-        true,
-        true,
-        true,
-        true,
-        false,
-        false,
-        false,
-        false,
-        false,
-        false,
-        null,
-        null,
-        null,
-        null,
-        null,
-        null
-      ]
+      "nullable": []
     }
   },
   "254234fc9c471f3012eaf52ac8bd18308710e2b0675de50bd433d28768997d45": {
@@ -1348,6 +1379,26 @@
           "Uuid",
           "Uuid",
           "Timestamptz",
+          "Int2"
+        ]
+      },
+      "nullable": []
+    }
+  },
+  "457d3b5ddc3734e94b4a4d254cd7f256fa7b09faed0d8361ce28503547161fe7": {
+    "query": "\nupdate jig\nset display_name     = coalesce($2, display_name),\n    author_id        = coalesce($3, author_id),\n    language         = coalesce($4, language),\n    description      = coalesce($5, description),\n    is_public        = coalesce($6, is_public),\n    direction        = coalesce($7, direction),\n    display_score    = coalesce($8, display_score),\n    theme            = coalesce($9, theme),\n    updated_at       = now()\nwhere id = $1\n  and (($2::text is not null and $2 is distinct from display_name) or\n       ($3::uuid is not null and $3 is distinct from author_id) or\n       ($4::text is not null and $4 is distinct from language) or\n       ($5::text is not null and $5 is distinct from description) or\n       ($6::bool is not null and $6 is distinct from is_public) or\n       ($7::smallint is not null and $7 is distinct from direction) or\n       ($8::bool is not null and $8 is distinct from display_score) or\n       ($9::smallint is not null and $9 is distinct from theme))\n",
+    "describe": {
+      "columns": [],
+      "parameters": {
+        "Left": [
+          "Uuid",
+          "Text",
+          "Uuid",
+          "Text",
+          "Text",
+          "Bool",
+          "Int2",
+          "Bool",
           "Int2"
         ]
       },
@@ -2083,6 +2134,146 @@
       ]
     }
   },
+  "73b93b6a269a7dec3b8fc929701ed4cb720523847dad00136288ade07321ffe3": {
+    "query": "\nselect  \n    id as \"id: JigId\",\n    display_name,\n    creator_id,\n    author_id,\n    publish_at,\n    updated_at,\n    language,\n    description,\n    is_public,\n    direction as \"direction: TextDirection\",\n    display_score,\n    theme as \"theme: ThemeId\",\n    audio_background as \"audio_background!: Option<AudioBackground>\",\n    array(select row(unnest(audio_feedback_positive))) as \"audio_feedback_positive!: Vec<(AudioFeedbackPositive,)>\", -- TODO: fix ugly!\n    array(select row(unnest(audio_feedback_negative))) as \"audio_feedback_negative!: Vec<(AudioFeedbackNegative,)>\",\n    array(\n        select row (id, kind)\n        from jig_module\n        where jig_id = $1\n        order by \"index\"\n    ) as \"modules!: Vec<(ModuleId, ModuleKind)>\",\n    array(select row(goal_id) from jig_goal where jig_id = $1) as \"goals!: Vec<(GoalId,)>\",\n    array(select row(category_id) from jig_category where jig_id = $1) as \"categories!: Vec<(CategoryId,)>\",\n    array(select row(affiliation_id) from jig_affiliation where jig_id = jig.id) as \"affiliations!: Vec<(AffiliationId,)>\",\n    array(select row(age_range_id) from jig_age_range where jig_id = jig.id) as \"age_ranges!: Vec<(AgeRangeId,)>\",\n    array(select row(id) from jig_additional_resource where jig_id = $1) as \"additional_resources!: Vec<(AdditionalResourceId,)>\"\nfrom jig\nwhere id = $1",
+    "describe": {
+      "columns": [
+        {
+          "ordinal": 0,
+          "name": "id: JigId",
+          "type_info": "Uuid"
+        },
+        {
+          "ordinal": 1,
+          "name": "display_name",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 2,
+          "name": "creator_id",
+          "type_info": "Uuid"
+        },
+        {
+          "ordinal": 3,
+          "name": "author_id",
+          "type_info": "Uuid"
+        },
+        {
+          "ordinal": 4,
+          "name": "publish_at",
+          "type_info": "Timestamptz"
+        },
+        {
+          "ordinal": 5,
+          "name": "updated_at",
+          "type_info": "Timestamptz"
+        },
+        {
+          "ordinal": 6,
+          "name": "language",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 7,
+          "name": "description",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 8,
+          "name": "is_public",
+          "type_info": "Bool"
+        },
+        {
+          "ordinal": 9,
+          "name": "direction: TextDirection",
+          "type_info": "Int2"
+        },
+        {
+          "ordinal": 10,
+          "name": "display_score",
+          "type_info": "Bool"
+        },
+        {
+          "ordinal": 11,
+          "name": "theme: ThemeId",
+          "type_info": "Int2"
+        },
+        {
+          "ordinal": 12,
+          "name": "audio_background!: Option<AudioBackground>",
+          "type_info": "Int2"
+        },
+        {
+          "ordinal": 13,
+          "name": "audio_feedback_positive!: Vec<(AudioFeedbackPositive,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 14,
+          "name": "audio_feedback_negative!: Vec<(AudioFeedbackNegative,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 15,
+          "name": "modules!: Vec<(ModuleId, ModuleKind)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 16,
+          "name": "goals!: Vec<(GoalId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 17,
+          "name": "categories!: Vec<(CategoryId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 18,
+          "name": "affiliations!: Vec<(AffiliationId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 19,
+          "name": "age_ranges!: Vec<(AgeRangeId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 20,
+          "name": "additional_resources!: Vec<(AdditionalResourceId,)>",
+          "type_info": "RecordArray"
+        }
+      ],
+      "parameters": {
+        "Left": [
+          "Uuid"
+        ]
+      },
+      "nullable": [
+        false,
+        false,
+        true,
+        true,
+        true,
+        true,
+        false,
+        false,
+        false,
+        false,
+        false,
+        false,
+        true,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null
+      ]
+    }
+  },
   "74b512fde797ad5d92d4a9fca1ce7d546e799815ed95169ad0591de8db129393": {
     "query": "\nselect user_id\nfrom user_auth_basic\nwhere\n    email = $1::text and\n    not exists(select 1 from user_email where email = $1)\n",
     "describe": {
@@ -2120,6 +2311,146 @@
       },
       "nullable": [
         false
+      ]
+    }
+  },
+  "763b624c44e2aa3d267be1fc384f4f066c45f9ea71d5a9ee18f8a215a7fe4b30": {
+    "query": "\nselect\n    id as \"id: JigId\",\n    display_name,\n    creator_id,\n    author_id,\n    publish_at,\n    updated_at,\n    language,\n    description,\n    is_public,\n    direction as \"direction: TextDirection\",\n    display_score,\n    theme as \"theme: ThemeId\",\n    audio_background as \"audio_background!: Option<AudioBackground>\",\n    array(select row(unnest(audio_feedback_positive))) as \"audio_feedback_positive!: Vec<(AudioFeedbackPositive,)>\", -- TODO: fix ugly!\n    array(select row(unnest(audio_feedback_negative))) as \"audio_feedback_negative!: Vec<(AudioFeedbackNegative,)>\",\n    array(\n        select row (id, kind)\n        from jig_module\n        where jig_id = jig.id\n        order by \"index\"\n    ) as \"modules!: Vec<(ModuleId, ModuleKind)>\",\n    array(select row(goal_id) from jig_goal where jig_id = jig.id) as \"goals!: Vec<(GoalId,)>\",\n    array(select row(category_id) from jig_category where jig_id = jig.id) as \"categories!: Vec<(CategoryId,)>\",\n    array(select row(affiliation_id) from jig_affiliation where jig_id = jig.id) as \"affiliations!: Vec<(AffiliationId,)>\",\n    array(select row(age_range_id) from jig_age_range where jig_id = jig.id) as \"age_ranges!: Vec<(AgeRangeId,)>\",\n    array(select row(id) from jig_additional_resource where jig_id = jig.id) as \"additional_resources!: Vec<(AdditionalResourceId,)>\"\nfrom jig\ninner join unnest($1::uuid[]) with ordinality t(id, ord) USING (id)\norder by t.ord\n",
+    "describe": {
+      "columns": [
+        {
+          "ordinal": 0,
+          "name": "id: JigId",
+          "type_info": "Uuid"
+        },
+        {
+          "ordinal": 1,
+          "name": "display_name",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 2,
+          "name": "creator_id",
+          "type_info": "Uuid"
+        },
+        {
+          "ordinal": 3,
+          "name": "author_id",
+          "type_info": "Uuid"
+        },
+        {
+          "ordinal": 4,
+          "name": "publish_at",
+          "type_info": "Timestamptz"
+        },
+        {
+          "ordinal": 5,
+          "name": "updated_at",
+          "type_info": "Timestamptz"
+        },
+        {
+          "ordinal": 6,
+          "name": "language",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 7,
+          "name": "description",
+          "type_info": "Text"
+        },
+        {
+          "ordinal": 8,
+          "name": "is_public",
+          "type_info": "Bool"
+        },
+        {
+          "ordinal": 9,
+          "name": "direction: TextDirection",
+          "type_info": "Int2"
+        },
+        {
+          "ordinal": 10,
+          "name": "display_score",
+          "type_info": "Bool"
+        },
+        {
+          "ordinal": 11,
+          "name": "theme: ThemeId",
+          "type_info": "Int2"
+        },
+        {
+          "ordinal": 12,
+          "name": "audio_background!: Option<AudioBackground>",
+          "type_info": "Int2"
+        },
+        {
+          "ordinal": 13,
+          "name": "audio_feedback_positive!: Vec<(AudioFeedbackPositive,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 14,
+          "name": "audio_feedback_negative!: Vec<(AudioFeedbackNegative,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 15,
+          "name": "modules!: Vec<(ModuleId, ModuleKind)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 16,
+          "name": "goals!: Vec<(GoalId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 17,
+          "name": "categories!: Vec<(CategoryId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 18,
+          "name": "affiliations!: Vec<(AffiliationId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 19,
+          "name": "age_ranges!: Vec<(AgeRangeId,)>",
+          "type_info": "RecordArray"
+        },
+        {
+          "ordinal": 20,
+          "name": "additional_resources!: Vec<(AdditionalResourceId,)>",
+          "type_info": "RecordArray"
+        }
+      ],
+      "parameters": {
+        "Left": [
+          "UuidArray"
+        ]
+      },
+      "nullable": [
+        false,
+        false,
+        true,
+        true,
+        true,
+        true,
+        false,
+        false,
+        false,
+        false,
+        false,
+        false,
+        true,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null,
+        null
       ]
     }
   },
@@ -2760,128 +3091,6 @@
       ]
     }
   },
-  "a711efdb60c95b359f7baa0d8ea9dd55a672e9dc39fb996d10c34db194af2880": {
-    "query": "\nselect  \n    id as \"id: JigId\",\n    display_name,\n    creator_id,\n    author_id,\n    publish_at,\n    updated_at,\n    language,\n    description,\n    is_public,\n    direction as \"direction: TextDirection\",\n    display_score,\n    theme as \"theme: ThemeId\",\n    array(\n        select row (id, kind)\n        from jig_module\n        where jig_id = jig.id\n        order by \"index\"\n    ) as \"modules!: Vec<(ModuleId, ModuleKind)>\",\n    array(select row(goal_id) from jig_goal where jig_id = jig.id) as \"goals!: Vec<(GoalId,)>\",\n    array(select row(category_id) from jig_category where jig_id = jig.id) as \"categories!: Vec<(CategoryId,)>\",\n    array(select row(affiliation_id) from jig_affiliation where jig_id = jig.id) as \"affiliations!: Vec<(AffiliationId,)>\",\n    array(select row(age_range_id) from jig_age_range where jig_id = jig.id) as \"age_ranges!: Vec<(AgeRangeId,)>\",\n    array(select row(id) from jig_additional_resource where jig_id = jig.id) as \"additional_resources!: Vec<(AdditionalResourceId,)>\"\nfrom jig\ninner join unnest($1::uuid[]) with ordinality t(id, ord) USING (id)\norder by t.ord\n",
-    "describe": {
-      "columns": [
-        {
-          "ordinal": 0,
-          "name": "id: JigId",
-          "type_info": "Uuid"
-        },
-        {
-          "ordinal": 1,
-          "name": "display_name",
-          "type_info": "Text"
-        },
-        {
-          "ordinal": 2,
-          "name": "creator_id",
-          "type_info": "Uuid"
-        },
-        {
-          "ordinal": 3,
-          "name": "author_id",
-          "type_info": "Uuid"
-        },
-        {
-          "ordinal": 4,
-          "name": "publish_at",
-          "type_info": "Timestamptz"
-        },
-        {
-          "ordinal": 5,
-          "name": "updated_at",
-          "type_info": "Timestamptz"
-        },
-        {
-          "ordinal": 6,
-          "name": "language",
-          "type_info": "Text"
-        },
-        {
-          "ordinal": 7,
-          "name": "description",
-          "type_info": "Text"
-        },
-        {
-          "ordinal": 8,
-          "name": "is_public",
-          "type_info": "Bool"
-        },
-        {
-          "ordinal": 9,
-          "name": "direction: TextDirection",
-          "type_info": "Int2"
-        },
-        {
-          "ordinal": 10,
-          "name": "display_score",
-          "type_info": "Bool"
-        },
-        {
-          "ordinal": 11,
-          "name": "theme: ThemeId",
-          "type_info": "Int2"
-        },
-        {
-          "ordinal": 12,
-          "name": "modules!: Vec<(ModuleId, ModuleKind)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 13,
-          "name": "goals!: Vec<(GoalId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 14,
-          "name": "categories!: Vec<(CategoryId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 15,
-          "name": "affiliations!: Vec<(AffiliationId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 16,
-          "name": "age_ranges!: Vec<(AgeRangeId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 17,
-          "name": "additional_resources!: Vec<(AdditionalResourceId,)>",
-          "type_info": "RecordArray"
-        }
-      ],
-      "parameters": {
-        "Left": [
-          "UuidArray"
-        ]
-      },
-      "nullable": [
-        false,
-        false,
-        true,
-        true,
-        true,
-        true,
-        false,
-        false,
-        false,
-        false,
-        false,
-        false,
-        null,
-        null,
-        null,
-        null,
-        null,
-        null
-      ]
-    }
-  },
   "a87ea4169f26518cbf7f88c1777bd15bbf72f938d6352567cd549bbe970db5f2": {
     "query": "\nupdate user_color\nset index = index - 1\nwhere index > $2 and user_id = $1\n",
     "describe": {
@@ -2979,6 +3188,20 @@
       "nullable": [
         false
       ]
+    }
+  },
+  "b3ced9e053666b805804107a80e32f6e8612a8668dec205803c6b50c6bca241f": {
+    "query": "\nupdate jig\nset audio_feedback_positive = $2,\n    audio_feedback_negative = $3,\n    updated_at = now()\nwhere id = $1 and ($2 <> audio_feedback_positive or $3 <> audio_feedback_negative)\n            ",
+    "describe": {
+      "columns": [],
+      "parameters": {
+        "Left": [
+          "Uuid",
+          "Int2Array",
+          "Int2Array"
+        ]
+      },
+      "nullable": []
     }
   },
   "b73bc1e83d2008fc5b9cc7e9a6c9a6b67b136c45a41e39a8929b554dc5c98485": {
@@ -3358,26 +3581,6 @@
       ]
     }
   },
-  "d6b9dde76778007181c50917cd19d80b5583fda8c33a71baae4114681f074718": {
-    "query": "\nupdate jig\nset display_name    = coalesce($2, display_name),\n    author_id       = coalesce($3, author_id),\n    language        = coalesce($4, language),\n    description     = coalesce($5, description),\n    is_public       = coalesce($6, is_public),\n    direction       = coalesce($7, direction),\n    display_score   = coalesce($8, display_score),\n    theme           = coalesce($9, theme),\n    updated_at      = now()\nwhere id = $1\n  and (($2::text is not null and $2 is distinct from display_name) or\n       ($3::uuid is not null and $3 is distinct from author_id) or\n       ($4::text is not null and $4 is distinct from language) or\n       ($5::text is not null and $5 is distinct from description) or\n       ($6::bool is not null and $6 is distinct from is_public) or\n       ($7::smallint is not null and $7 is distinct from direction) or\n       ($8::bool is not null and $8 is distinct from display_score) or\n       ($9::smallint is not null and $9 is distinct from theme))",
-    "describe": {
-      "columns": [],
-      "parameters": {
-        "Left": [
-          "Uuid",
-          "Text",
-          "Uuid",
-          "Text",
-          "Text",
-          "Bool",
-          "Int2",
-          "Bool",
-          "Int2"
-        ]
-      },
-      "nullable": []
-    }
-  },
   "e13fed8948ad863f394fefbda8a3b5a002208ed80892e3a5751742eb6d938815": {
     "query": "\nselect user_id as \"id\",\n    username,\n    user_email.email::text                                                              as \"email!\",\n    given_name,\n    family_name,\n    language,\n    locale,\n    opt_into_edu_resources,\n    over_18,\n    timezone,\n    user_profile.created_at,\n    user_profile.updated_at,\n    organization,\n    location,\n    array(select scope from user_scope where user_scope.user_id = \"user\".id) as \"scopes!: Vec<i16>\",\n    array(select subject_id from user_subject where user_subject.user_id = \"user\".id) as \"subjects!: Vec<Uuid>\",\n    array(select affiliation_id from user_affiliation where user_affiliation.user_id = \"user\".id) as \"affiliations!: Vec<Uuid>\",\n    array(select age_range_id from user_age_range where user_age_range.user_id = \"user\".id) as \"age_ranges!: Vec<Uuid>\"\nfrom \"user\"\ninner join user_profile on \"user\".id = user_profile.user_id\ninner join user_email using(user_id)\nwhere id = $1",
     "describe": {
@@ -3511,128 +3714,6 @@
         ]
       },
       "nullable": []
-    }
-  },
-  "e288e2e294f9fd29c9d6cd46be2172fc1e766a8da098227d84cd87c6ed137c95": {
-    "query": "\nselect  \n    id as \"id: JigId\",\n    display_name,\n    creator_id,\n    author_id,\n    publish_at,\n    updated_at,\n    language,\n    description,\n    is_public,\n    direction as \"direction: TextDirection\",\n    display_score,\n    theme as \"theme: ThemeId\",\n    array(\n        select row (id, kind)\n        from jig_module\n        where jig_id = $1\n        order by \"index\"\n    ) as \"modules!: Vec<(ModuleId, ModuleKind)>\",\n    array(select row(goal_id) from jig_goal where jig_id = $1) as \"goals!: Vec<(GoalId,)>\",\n    array(select row(category_id) from jig_category where jig_id = $1) as \"categories!: Vec<(CategoryId,)>\",\n    array(select row(affiliation_id) from jig_affiliation where jig_id = jig.id) as \"affiliations!: Vec<(AffiliationId,)>\",\n    array(select row(age_range_id) from jig_age_range where jig_id = jig.id) as \"age_ranges!: Vec<(AgeRangeId,)>\",\n    array(select row(id) from jig_additional_resource where jig_id = $1) as \"additional_resources!: Vec<(AdditionalResourceId,)>\"\nfrom jig\nwhere id = $1",
-    "describe": {
-      "columns": [
-        {
-          "ordinal": 0,
-          "name": "id: JigId",
-          "type_info": "Uuid"
-        },
-        {
-          "ordinal": 1,
-          "name": "display_name",
-          "type_info": "Text"
-        },
-        {
-          "ordinal": 2,
-          "name": "creator_id",
-          "type_info": "Uuid"
-        },
-        {
-          "ordinal": 3,
-          "name": "author_id",
-          "type_info": "Uuid"
-        },
-        {
-          "ordinal": 4,
-          "name": "publish_at",
-          "type_info": "Timestamptz"
-        },
-        {
-          "ordinal": 5,
-          "name": "updated_at",
-          "type_info": "Timestamptz"
-        },
-        {
-          "ordinal": 6,
-          "name": "language",
-          "type_info": "Text"
-        },
-        {
-          "ordinal": 7,
-          "name": "description",
-          "type_info": "Text"
-        },
-        {
-          "ordinal": 8,
-          "name": "is_public",
-          "type_info": "Bool"
-        },
-        {
-          "ordinal": 9,
-          "name": "direction: TextDirection",
-          "type_info": "Int2"
-        },
-        {
-          "ordinal": 10,
-          "name": "display_score",
-          "type_info": "Bool"
-        },
-        {
-          "ordinal": 11,
-          "name": "theme: ThemeId",
-          "type_info": "Int2"
-        },
-        {
-          "ordinal": 12,
-          "name": "modules!: Vec<(ModuleId, ModuleKind)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 13,
-          "name": "goals!: Vec<(GoalId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 14,
-          "name": "categories!: Vec<(CategoryId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 15,
-          "name": "affiliations!: Vec<(AffiliationId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 16,
-          "name": "age_ranges!: Vec<(AgeRangeId,)>",
-          "type_info": "RecordArray"
-        },
-        {
-          "ordinal": 17,
-          "name": "additional_resources!: Vec<(AdditionalResourceId,)>",
-          "type_info": "RecordArray"
-        }
-      ],
-      "parameters": {
-        "Left": [
-          "Uuid"
-        ]
-      },
-      "nullable": [
-        false,
-        false,
-        true,
-        true,
-        true,
-        true,
-        false,
-        false,
-        false,
-        false,
-        false,
-        false,
-        null,
-        null,
-        null,
-        null,
-        null,
-        null
-      ]
     }
   },
   "e3385b6d93728cc62d0995e839f0148aaba5e572c06049512a4f31d44f03c269": {

--- a/backend/api/src/db/jig.rs
+++ b/backend/api/src/db/jig.rs
@@ -5,7 +5,8 @@ use shared::domain::{
     jig::{
         additional_resource::AdditionalResourceId,
         module::{body::cover, ModuleBody, ModuleId},
-        Jig, JigId, LiteModule, ModuleKind, TextDirection,
+        AudioBackground, AudioEffects, AudioFeedbackNegative, AudioFeedbackPositive, Jig, JigId,
+        LiteModule, ModuleKind, TextDirection,
     },
     meta::{AffiliationId, AgeRangeId, GoalId},
     user::UserScope,
@@ -86,11 +87,13 @@ pub enum CreateJigError {
     Sqlx(sqlx::Error),
     DefaultModules(serde_json::Error),
 }
+
 impl From<sqlx::Error> for CreateJigError {
     fn from(e: sqlx::Error) -> Self {
         Self::Sqlx(e)
     }
 }
+
 impl From<serde_json::Error> for CreateJigError {
     fn from(e: serde_json::Error) -> Self {
         Self::DefaultModules(e)
@@ -98,9 +101,9 @@ impl From<serde_json::Error> for CreateJigError {
 }
 
 pub async fn get_by_ids(db: &PgPool, ids: &[Uuid]) -> sqlx::Result<Vec<Jig>> {
-    let v = sqlx::query!(
+    let v = sqlx::query!( //language=SQL
 r#"
-select  
+select
     id as "id: JigId",
     display_name,
     creator_id,
@@ -113,6 +116,9 @@ select
     direction as "direction: TextDirection",
     display_score,
     theme as "theme: ThemeId",
+    audio_background as "audio_background!: Option<AudioBackground>",
+    array(select row(unnest(audio_feedback_positive))) as "audio_feedback_positive!: Vec<(AudioFeedbackPositive,)>", -- TODO: fix ugly!
+    array(select row(unnest(audio_feedback_negative))) as "audio_feedback_negative!: Vec<(AudioFeedbackNegative,)>",
     array(
         select row (id, kind)
         from jig_module
@@ -128,7 +134,7 @@ from jig
 inner join unnest($1::uuid[]) with ordinality t(id, ord) USING (id)
 order by t.ord
 "#, ids)
-    .fetch_all(db).await?;
+        .fetch_all(db).await?;
 
     let v = v
         .into_iter()
@@ -159,6 +165,19 @@ order by t.ord
                 .into_iter()
                 .map(|(it,)| it)
                 .collect(),
+            audio_background: row.audio_background,
+            audio_effects: AudioEffects {
+                feedback_positive: row
+                    .audio_feedback_positive
+                    .into_iter()
+                    .map(|(it,)| it)
+                    .collect(),
+                feedback_negative: row
+                    .audio_feedback_negative
+                    .into_iter()
+                    .map(|(it,)| it)
+                    .collect(),
+            },
         })
         .collect();
 
@@ -166,7 +185,7 @@ order by t.ord
 }
 
 pub async fn get(pool: &PgPool, id: JigId) -> anyhow::Result<Option<Jig>> {
-    let jig = sqlx::query!(
+    let jig = sqlx::query!( //language=SQL
         r#"
 select  
     id as "id: JigId",
@@ -181,6 +200,9 @@ select
     direction as "direction: TextDirection",
     display_score,
     theme as "theme: ThemeId",
+    audio_background as "audio_background!: Option<AudioBackground>",
+    array(select row(unnest(audio_feedback_positive))) as "audio_feedback_positive!: Vec<(AudioFeedbackPositive,)>", -- TODO: fix ugly!
+    array(select row(unnest(audio_feedback_negative))) as "audio_feedback_negative!: Vec<(AudioFeedbackNegative,)>",
     array(
         select row (id, kind)
         from jig_module
@@ -196,32 +218,37 @@ from jig
 where id = $1"#,
         id.0
     )
-    .fetch_optional(pool)
-    .await?
-    .map(|row| Jig {
-        id: row.id,
-        display_name: row.display_name,
-        language: row.language,
-        modules: row
-            .modules
-            .into_iter()
-            .map(|(id, kind)| LiteModule { id, kind })
-            .collect(),
-        goals: row.goals.into_iter().map(|(it,)| it).collect(),
-        categories: row.categories.into_iter().map(|(it,)| it).collect(),
-        creator_id: row.creator_id,
-        author_id: row.author_id,
-        publish_at: row.publish_at,
-        description: row.description,
-        last_edited: row.updated_at,
-        is_public: row.is_public,
-        direction: row.direction,
-        display_score: row.display_score,
-        theme: row.theme,
-        age_ranges: row.age_ranges.into_iter().map(|(it,)| it).collect(),
-        affiliations: row.affiliations.into_iter().map(|(it,)| it).collect(),
-        additional_resources: row.additional_resources.into_iter().map(|(it,) | it).collect(),
-    });
+        .fetch_optional(pool)
+        .await?
+        .map(|row| Jig {
+            id: row.id,
+            display_name: row.display_name,
+            language: row.language,
+            modules: row
+                .modules
+                .into_iter()
+                .map(|(id, kind)| LiteModule { id, kind })
+                .collect(),
+            goals: row.goals.into_iter().map(|(it, )| it).collect(),
+            categories: row.categories.into_iter().map(|(it, )| it).collect(),
+            creator_id: row.creator_id,
+            author_id: row.author_id,
+            publish_at: row.publish_at,
+            description: row.description,
+            last_edited: row.updated_at,
+            is_public: row.is_public,
+            direction: row.direction,
+            display_score: row.display_score,
+            theme: row.theme,
+            age_ranges: row.age_ranges.into_iter().map(|(it, )| it).collect(),
+            affiliations: row.affiliations.into_iter().map(|(it, )| it).collect(),
+            additional_resources: row.additional_resources.into_iter().map(|(it, )| it).collect(),
+            audio_background: row.audio_background,
+            audio_effects: AudioEffects {
+                feedback_positive: row.audio_feedback_positive.into_iter().map(|(it,)| it).collect(),
+                feedback_negative: row.audio_feedback_negative.into_iter().map(|(it,)| it).collect(),
+            },
+        });
 
     Ok(jig)
 }
@@ -242,6 +269,8 @@ pub async fn update(
     direction: Option<TextDirection>,
     display_score: Option<bool>,
     theme: Option<ThemeId>,
+    audio_background: Option<Option<AudioBackground>>,
+    audio_effects: Option<AudioEffects>,
 ) -> Result<(), error::UpdateWithMetadata> {
     let mut transaction = pool.begin().await?;
     if !sqlx::query!(
@@ -268,19 +297,59 @@ where id = $1 and $2 is distinct from publish_at"#,
         .await?;
     }
 
+    if let Some(audio_background) = audio_background {
+        sqlx::query!(
+            r#"
+update jig
+set audio_background = $2, updated_at = now()
+where id = $1 and $2 is distinct from audio_background
+            "#,
+            id.0,
+            audio_background.map(|it| it as i16),
+        )
+        .execute(&mut transaction)
+        .await?;
+    }
+
+    if let Some(audio_effects) = audio_effects {
+        sqlx::query!(
+            //language=SQL
+            r#"
+update jig
+set audio_feedback_positive = $2,
+    audio_feedback_negative = $3,
+    updated_at = now()
+where id = $1 and ($2 <> audio_feedback_positive or $3 <> audio_feedback_negative)
+            "#,
+            id.0,
+            &audio_effects
+                .feedback_positive
+                .into_iter()
+                .map(|it| it as i16)
+                .collect::<Vec<_>>(),
+            &audio_effects
+                .feedback_negative
+                .into_iter()
+                .map(|it| it as i16)
+                .collect::<Vec<_>>(),
+        )
+        .execute(&mut transaction)
+        .await?;
+    }
+
     sqlx::query!(
         //language=SQL
         r#"
 update jig
-set display_name    = coalesce($2, display_name),
-    author_id       = coalesce($3, author_id),
-    language        = coalesce($4, language),
-    description     = coalesce($5, description),
-    is_public       = coalesce($6, is_public),
-    direction       = coalesce($7, direction),
-    display_score   = coalesce($8, display_score),
-    theme           = coalesce($9, theme),
-    updated_at      = now()
+set display_name     = coalesce($2, display_name),
+    author_id        = coalesce($3, author_id),
+    language         = coalesce($4, language),
+    description      = coalesce($5, description),
+    is_public        = coalesce($6, is_public),
+    direction        = coalesce($7, direction),
+    display_score    = coalesce($8, display_score),
+    theme            = coalesce($9, theme),
+    updated_at       = now()
 where id = $1
   and (($2::text is not null and $2 is distinct from display_name) or
        ($3::uuid is not null and $3 is distinct from author_id) or
@@ -289,7 +358,8 @@ where id = $1
        ($6::bool is not null and $6 is distinct from is_public) or
        ($7::smallint is not null and $7 is distinct from direction) or
        ($8::bool is not null and $8 is distinct from display_score) or
-       ($9::smallint is not null and $9 is distinct from theme))"#,
+       ($9::smallint is not null and $9 is distinct from theme))
+"#,
         id.0,
         display_name,
         author_id,
@@ -361,6 +431,9 @@ select
     direction as "direction: TextDirection",
     display_score,
     theme as "theme: ThemeId",
+    audio_background as "audio_background!: Option<AudioBackground>",
+    array(select row(unnest(audio_feedback_positive))) as "audio_feedback_positive!: Vec<(AudioFeedbackPositive,)>", -- TODO: fix ugly!
+    array(select row(unnest(audio_feedback_negative))) as "audio_feedback_negative!: Vec<(AudioFeedbackNegative,)>",
     array(
         select row (id, kind)
         from jig_module
@@ -383,33 +456,38 @@ limit 20 offset 20 * $2
         page,
         author_id,
     )
-    .fetch(pool)
-    .map_ok(|row| Jig {
-        id: row.id,
-        display_name: row.display_name,
-        language: row.language,
-        modules: row
-            .modules
-            .into_iter()
-            .map(|(id, kind)| LiteModule { id, kind })
-            .collect(),
-        goals: row.goals.into_iter().map(|(it,)| it).collect(),
-        categories: row.categories.into_iter().map(|(it,)| it).collect(),
-        creator_id: row.creator_id,
-        author_id: row.author_id,
-        publish_at: row.publish_at,
-        description: row.description,
-        last_edited: row.updated_at,
-        is_public: row.is_public,
-        direction: row.direction,
-        display_score: row.display_score,
-        theme: row.theme,
-        age_ranges: row.age_ranges.into_iter().map(|(it,)| it).collect(),
-        affiliations: row.affiliations.into_iter().map(|(it,)| it).collect(),
-        additional_resources: row.additional_resources.into_iter().map(|(it,) | it).collect(),
-    })
-    .try_collect()
-    .await
+        .fetch(pool)
+        .map_ok(|row| Jig {
+            id: row.id,
+            display_name: row.display_name,
+            language: row.language,
+            modules: row
+                .modules
+                .into_iter()
+                .map(|(id, kind)| LiteModule { id, kind })
+                .collect(),
+            goals: row.goals.into_iter().map(|(it, )| it).collect(),
+            categories: row.categories.into_iter().map(|(it, )| it).collect(),
+            creator_id: row.creator_id,
+            author_id: row.author_id,
+            publish_at: row.publish_at,
+            description: row.description,
+            last_edited: row.updated_at,
+            is_public: row.is_public,
+            direction: row.direction,
+            display_score: row.display_score,
+            theme: row.theme,
+            age_ranges: row.age_ranges.into_iter().map(|(it, )| it).collect(),
+            affiliations: row.affiliations.into_iter().map(|(it, )| it).collect(),
+            additional_resources: row.additional_resources.into_iter().map(|(it, )| it).collect(),
+            audio_background: row.audio_background,
+            audio_effects: AudioEffects {
+                feedback_positive: row.audio_feedback_positive.into_iter().map(|(it, )| it).collect(),
+                feedback_negative: row.audio_feedback_negative.into_iter().map(|(it, )| it).collect(),
+            },
+        })
+        .try_collect()
+        .await
 }
 
 pub async fn filtered_count(
@@ -438,8 +516,20 @@ pub async fn clone(db: &PgPool, parent: JigId, user_id: Uuid) -> sqlx::Result<Op
 
     let new_id = sqlx::query!( //language=SQL
         r#"
-insert into jig (display_name, parents, creator_id, author_id, language, description, direction, display_score, theme)
-select display_name, array_append(parents, id), $2 as creator_id, $2 as author_id, language, description, direction, display_score, theme
+insert into jig (display_name, parents, creator_id, author_id, language, description, direction, display_score, theme,
+                 audio_background, audio_feedback_positive, audio_feedback_negative)
+select display_name,
+       array_append(parents, id),
+       $2 as creator_id,
+       $2 as author_id,
+       language,
+       description,
+       direction,
+       display_score,
+       theme,
+       audio_background,
+       audio_feedback_positive,
+       audio_feedback_negative
 from jig
 where id = $1
 returning id

--- a/backend/api/src/http/endpoints/jig.rs
+++ b/backend/api/src/http/endpoints/jig.rs
@@ -139,6 +139,8 @@ async fn update(
         req.direction,
         req.display_score,
         req.theme,
+        req.audio_background,
+        req.audio_effects,
     )
     .await?;
 

--- a/backend/api/tests/integration/jig.rs
+++ b/backend/api/tests/integration/jig.rs
@@ -137,7 +137,14 @@ async fn clone() -> anyhow::Result<()> {
 
     app.stop(false).await;
 
-    insta::assert_json_snapshot!(body, {".**.id" => "[id]", ".**.last_edited" => "[last_edited]"});
+    insta::assert_json_snapshot!(
+        body, {
+            ".**.id" => "[id]",
+            ".**.last_edited" => "[last_edited]",
+            ".**.feedback_positive" => "[audio]",
+            ".**.feedback_negative" => "[audio]"
+        }
+    );
 
     Ok(())
 }
@@ -166,7 +173,13 @@ async fn get() -> anyhow::Result<()> {
 
     app.stop(false).await;
 
-    insta::assert_json_snapshot!(body, {".**.last_edited" => "[last_edited]"});
+    insta::assert_json_snapshot!(
+        body, {
+            ".**.last_edited" => "[last_edited]",
+            ".**.feedback_positive" => "[audio]",
+            ".**.feedback_negative" => "[audio]"
+        }
+    );
 
     Ok(())
 }
@@ -193,7 +206,13 @@ async fn browse_simple() -> anyhow::Result<()> {
 
     app.stop(false).await;
 
-    insta::assert_json_snapshot!(body, {".**.last_edited" => "[last_edited]"});
+    insta::assert_json_snapshot!(
+        body, {
+            ".**.last_edited" => "[last_edited]",
+            ".**.feedback_positive" => "[audio]",
+            ".**.feedback_negative" => "[audio]"
+        }
+    );
 
     Ok(())
 }
@@ -223,7 +242,13 @@ async fn browse_own_simple() -> anyhow::Result<()> {
 
     app.stop(false).await;
 
-    insta::assert_json_snapshot!(body, {".**.last_edited" => "[last_edited]"});
+    insta::assert_json_snapshot!(
+        body, {
+            ".**.last_edited" => "[last_edited]",
+            ".**.feedback_positive" => "[audio]",
+            ".**.feedback_negative" => "[audio]"
+        }
+    );
 
     Ok(())
 }

--- a/backend/api/tests/integration/jig/cover.rs
+++ b/backend/api/tests/integration/jig/cover.rs
@@ -21,7 +21,6 @@ async fn update_no_modules_changes() -> anyhow::Result<()> {
         ))
         .json(&json! {{
             "display_name": "test",
-
         }})
         .login()
         .send()
@@ -42,7 +41,7 @@ async fn update_no_modules_changes() -> anyhow::Result<()> {
 
     app.stop(false).await;
 
-    insta::assert_json_snapshot!(body.jig, {".**.last_edited" => "[last_edited]"});
+    insta::assert_json_snapshot!(body.jig, {".**.last_edited" => "[last_edited]", ".**.feedback_positive" => "[audio]", ".**.feedback_negative" => "[audio]"});
 
     Ok(())
 }

--- a/shared/rust/src/domain/jig.rs
+++ b/shared/rust/src/domain/jig.rs
@@ -14,6 +14,7 @@ use chrono::{DateTime, Utc};
 #[cfg(feature = "backend")]
 use paperclip::actix::Apiv2Schema;
 use serde::{Deserialize, Serialize};
+use std::collections::HashSet;
 use uuid::Uuid;
 
 // avoid breaking Changes
@@ -21,6 +22,7 @@ pub use module::{LiteModule, Module, ModuleKind};
 
 use crate::domain::jig::module::body::ThemeId;
 pub use additional_resource::AdditionalResourceId;
+use std::convert::TryFrom;
 
 /// Wrapper type around [`Uuid`], represents the ID of a JIG.
 #[derive(Copy, Clone, Eq, PartialEq, Serialize, Deserialize, Debug)]
@@ -192,44 +194,79 @@ pub struct Jig {
 
     /// Theme for this jig, identified by `[ThemeId](jig::module::body::ThemeId)`.
     pub theme: ThemeId,
-    // TODO: need Audio enums to exist
-    /// Background audio
-    pub audio_bg: Option<AudioBackground>,
 
-    /// Audio effects 
+    /// Background audio
+    pub audio_background: Option<AudioBackground>,
+
+    /// Audio effects
     pub audio_effects: AudioEffects,
 }
 
-/// Audio for background music 
+/// Audio for background music
 #[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq)]
+#[cfg_attr(feature = "backend", derive(sqlx::Type))]
 #[cfg_attr(feature = "backend", derive(Apiv2Schema))]
+#[repr(i16)]
 pub enum AudioBackground {
+    #[allow(missing_docs)]
+    Placeholder0 = 0,
+    #[allow(missing_docs)]
+    Placeholder1 = 1,
+    #[allow(missing_docs)]
+    Placeholder2 = 2,
+    #[allow(missing_docs)]
+    Placeholder3 = 3,
+    #[allow(missing_docs)]
+    Placeholder4 = 4,
 }
 
 /// Audio Effects
 #[derive(Serialize, Deserialize, Debug, Clone, Default)]
 #[cfg_attr(feature = "backend", derive(Apiv2Schema))]
 pub struct AudioEffects {
-    /// Positive audio feedback 
-    pub feedback_positive: std::collections::HashSet<AudioFeedbackPositive>,
+    /// Positive audio feedback
+    pub feedback_positive: HashSet<AudioFeedbackPositive>,
 
-    /// Negative audio feedback 
-    pub feedback_negative: std::collections::HashSet<AudioFeedbackNegative>,
+    /// Negative audio feedback
+    pub feedback_negative: HashSet<AudioFeedbackNegative>,
 }
 
 /// Negative Audio Feedback
 #[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "backend", derive(sqlx::Type))]
 #[cfg_attr(feature = "backend", derive(Apiv2Schema))]
+#[repr(i16)]
 pub enum AudioFeedbackNegative {
-    
+    #[allow(missing_docs)]
+    Placeholder0 = 0,
+    #[allow(missing_docs)]
+    Placeholder1 = 1,
+    #[allow(missing_docs)]
+    Placeholder2 = 2,
+    #[allow(missing_docs)]
+    Placeholder3 = 3,
+    #[allow(missing_docs)]
+    Placeholder4 = 4,
 }
 
 /// Positive Audio Feedback
 #[derive(Serialize, Deserialize, Debug, Clone, Eq, PartialEq, Hash)]
+#[cfg_attr(feature = "backend", derive(sqlx::Type))]
 #[cfg_attr(feature = "backend", derive(Apiv2Schema))]
+#[repr(i16)]
 pub enum AudioFeedbackPositive {
-
+    #[allow(missing_docs)]
+    Placeholder0 = 0,
+    #[allow(missing_docs)]
+    Placeholder1 = 1,
+    #[allow(missing_docs)]
+    Placeholder2 = 2,
+    #[allow(missing_docs)]
+    Placeholder3 = 3,
+    #[allow(missing_docs)]
+    Placeholder4 = 4,
 }
+
 /// The response returned when a request for `GET`ing a jig is successful.
 #[derive(Serialize, Deserialize, Debug, Clone)]
 #[cfg_attr(feature = "backend", derive(Apiv2Schema))]
@@ -314,22 +351,17 @@ pub struct JigUpdateRequest {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[serde(default)]
     pub theme: Option<ThemeId>,
-    // TODO: need Audio enums to exist
-    // /// Background audio
-    // #[serde(skip_serializing_if = "Option::is_none")]
-    // #[serde(deserialize_with = "super::deserialize_optional_field")]
-    // #[serde(default)]
-    // pub background_audio: Option<Option<BackgroundAudio>>,
 
-    // /// Correct answer audio
-    // #[serde(skip_serializing_if = "HashSet::is_empty")]
-    // #[serde(default)]
-    // pub correct_answer_audio: Option<HashSet<FeedbackAudio>>,
+    /// Background audio
+    #[serde(deserialize_with = "super::deserialize_optional_field")]
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub audio_background: Option<Option<AudioBackground>>,
 
-    // /// Wrong answer audio
-    // #[serde(skip_serializing_if = "HashSet::is_empty")]
-    // #[serde(default)]
-    // pub wrong_answer_audio: Option<HashSet<FeedbackAudio>>,
+    /// Audio effects
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(default)]
+    pub audio_effects: Option<AudioEffects>,
 }
 
 /// Query for [`Browse`](crate::api::endpoints::jig::Browse).

--- a/shared/rust/src/domain/jig/module/body.rs
+++ b/shared/rust/src/domain/jig/module/body.rs
@@ -1,5 +1,5 @@
 use crate::{
-    domain::{audio::AudioId, image::ImageId, jig::Jig},
+    domain::{audio::AudioId, image::ImageId},
     media::MediaLibrary,
 };
 #[cfg(feature = "backend")]

